src/Streams/StreamReader.h

```diff
@@ -39,4 +41,18 @@ class StreamReader {
 	inline void Read(std::vector<T, A>& vector) {
 		ReadImplementation(vector.data(), vector.size() * sizeof(T));
 	}
+
+	// std::string prefixed by the string's size. The type of integer representing the size must be provided. 
+	template<typename SizeType>
+	void Read(std::string& string) {
+		SizeType stringSize;
+		Read(stringSize);
+		if (stringSize < 0) {
```

Comment:
I wonder if there is a way to eliminate this check if `SizeType` is `unsigned`.

What if `SizeType` is greater than `std::size_t`, such as a `uint64_t` on a 32-bit build.